### ISSUE TITLE
Shaping and display of certain language is completely incorrect.

### ISSUE THREAD

**Issue by milizhang** [microsoft/terminal#4708]
# Environment

```none
Windows build number: Windows 10 2004 (10.0.19041.84)
Windows Terminal version (if applicable): master (at b8e33560f90ff7c4b3294f83bbeabb1b6a74d3e1)
```

# Steps to reproduce
- Download [test file](https://github.com/microsoft/terminal/files/4121097/utf82.txt) from #4375.
- Open the file in Windows notepad.txt and resave it in UTF-16LE encoding. 
- Using `type utf82.txt` from Windows `cmd` to show the content of the file.

# Expected behavior
![image](https://user-images.githubusercontent.com/1834979/75143001-10d3d000-56a9-11ea-8ab5-1c41adf5465e.png)

(This screenshot is from 0.9.433.0)

# Actual behavior
![image](https://user-images.githubusercontent.com/1834979/75142960-031e4a80-56a9-11ea-942a-b9f968d94c99.png)

**Comment by milizhang** [comment]
Hit this while trying to continue my work on #4375.
I have not got the time to analyze into it, but it seems to be caused by some change between 0.9.433 and the current master.

**Comment by zadjii-msft** [comment]
I'd bet money that this regressed with 4420950, which I think is also already being tracked in #4704

**Comment by DHowett-MSFT** [comment]
Yeah, this is definitely a /dup of #4704.

**Comment by miniksa** [comment]
I see what's wrong here and it's not going to be exactly the same fix as #4704. Splitting this back out.
